backend/otel/systemd.go

```diff
@@ -0,0 +1,40 @@
+package otel
+
+import (
+	"go.opentelemetry.io/collector/pdata/plog"
+	"strconv"
+)
+
+type SystemdKey = string
+
+const Message SystemdKey = "MESSAGE"
+const Priority SystemdKey = "PRIORITY"
+
+func extractSystemd(fields *extractedFields, m map[string]any) {
+	fields.logBody = m[Message].(string)
+	if priority, err := strconv.ParseInt(m[Priority].(string), 10, 4); err == nil {
+		switch priority {
+		case 0:
+			fields.logSeverity = plog.SeverityNumberFatal.String()
+		case 1:
+			fields.logSeverity = plog.SeverityNumberFatal.String()
+		case 2:
+			fields.logSeverity = plog.SeverityNumberError.String()
+		case 3:
+			fields.logSeverity = plog.SeverityNumberError.String()
+		case 4:
+			fields.logSeverity = plog.SeverityNumberWarn.String()
+		case 5:
+			fields.logSeverity = plog.SeverityNumberWarn.String()
+		case 6:
+			fields.logSeverity = plog.SeverityNumberInfo.String()
+		case 7:
+			fields.logSeverity = plog.SeverityNumberDebug.String()
```

Comment:
More of a curiosity, but is it possible to combine cases in go?
```
switch priority {
case 0, 1:
...
case 2, 3:
...
case 4, 5:
...
case 6:
...
case 7:
...
}
```

Reply:
TIL! ty